src/utils/getValidSubjectIconName.ts

```diff
@@ -1,11 +1,18 @@
 import { OakIconName, isValidIconName } from "@oaknational/oak-components";
 
+import { getValidSubjectCategoryIconById } from "./getValidSubjectCategoryIconById";
+
+import { SubjectCategory } from "@/utils/curriculum/types";
 import errorReporter from "@/common-lib/error-reporter";
 import OakError from "@/errors/OakError";
 
 export const getValidSubjectIconName = (
-  subject: string | null,
+  subject: string | null | number | SubjectCategory,
 ): OakIconName => {
+  if (subject && typeof subject === "object" && "id" in subject) {
+    return getValidSubjectCategoryIconById(subject.id) as OakIconName;
```

Comment:
Rather than pollute this function, I'd do it the other way around instead. So keep this "as-is" because it's a well defined function and we're adding a bit of a hack.

Make `getValidSubjectCategoryIconById` call `getValidSubjectIconName` instead.